### PR TITLE
Update DG for MarkApplicant Implementation

### PR DESCRIPTION
**Pull request by ro4k9**
- Add how the MarkApplicant parser parses the input. This is to be continued with the implementation of the command itself
- Add activity diagram of MarkApplicant command
- Add sequence diagram of MarkApplicant command

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -208,6 +208,37 @@ The following activity diagram summarizes what happens when a user executes a ne
...
 ![EditApplicantActivityDiagram](images/EditApplicantActivityDiagram.png)
 
+### MarkApplicant feature
+
+The `markapplicant` mechanism is facilitated by `AddressBook`. `MarkApplicant` extends `Command` class.
+`MarkApplicant` updates the status via creating a new applicant object with  the new status value
+and same original Applicant details.  This new Applicant object will replace the current Applicant object that is in the `AddressBook`.
+
+When the user wants to mark an applicant, the user will input `markapplicant` along with the index number of Applicant,
+and the new status value, which includes `pending`, `rejected`, and `accepted`
```
Comment (Justinnje) [inline docs/DeveloperGuide.md:218]: I think in the telegram discussion we came to a conclusion that this should have 4 states. Missing "interviewed"

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -208,6 +208,37 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+
+For example, `markapplicant 1 s/rejected` will change the status of applicant in index 1 to rejected.
```
Comment (Justinnje) [inline docs/DeveloperGuide.md:220]: Maybe you could also add that when a Applicant object is created, the initial value of the Applicant Status is set to "pending".

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -208,6 +208,37 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+Step 1. The user inputs `markapplicant 1 s/rejected` into ReCLIne. `AddressBookParser#parseCommand()`
+and `MarkApplicantParser#parse()`. `MarkApplicantParser#parse()`calls `ParserUtil#parseIndex()` and `ParserUtil#parseApplicant()`,
+which returns Index and ApplicantStatus objects representing the index and status value that user inputted. This returns a
+`MarkApplicant` with Index and ApplicantStatus objects as arguments.
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** If the `index` inputted is greater than the
+size of the current `UniqueApplicantList` the execution of the command will fail. A `CommandException` will be thrown
+and displayed for the user. This ensures that inputted `index` is not out of bound.
+</div>
+
```
Comment (Justinnje) [inline docs/DeveloperGuide.md:233]: You may want to add more steps, follow the activity diagram. So maybe like step 2 can be "a new Applicant Object is created with the new Applicant Status...."

**Review by Justinnje — COMMENTED** [review]
Some small things to change

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (ro4k9) [reply docs/DeveloperGuide.md:234]: I worked on the implementation with @chanyijuan so I worked on my part of the implementation of markapplicant :)

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (ro4k9) [reply docs/DeveloperGuide.md]: I updated it!

**Pull request closed without merging**